api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -485,4 +489,23 @@ public void syncEraCommonsStatusUsingImpersonation(User user) throws IOException
       }
     }
   }
+
+  public void syncTwoFactorAuthStatus() {
+    syncTwoFactorAuthStatus(userProvider.get());
+  }
+
+  public void syncTwoFactorAuthStatus(User targetUser) {
+    updateUserWithRetries(user -> {
+      if (user.getTwoFactorAuthCompletionTime() == null &&
```

Comment:
i think this actually unsets the user's 2fa completion time if it set to anything. `user.getTwoFactorAuthCompletionTime() = something` -> `user.getTwoFactorAuthCompletionTime() == null` is false -> flows into else statement.

might be good to write a test case to cover this as well as the previous condition. its not a lot of code its covering but boolean statements like this can be tricky so I think its good to have the edge cases covered.

Reply:
thanks for checking.... so here's what i think the logic should be -- since we need to be able to unset the var on our side if the user disables 2FA on the google side (and this method gets called by our cron endpoint that checks that daily to keep them in sync), it should check if the user is enrolled in google, then if the value is null in workbench, set it to a new timestamp.  otherwise (if user is unenrolled in google), set to null in workbench.